pubsub/drivertest/drivertest.go

```diff
@@ -583,27 +584,37 @@ func benchmark(b *testing.B, topic *pubsub.Topic, sub *pubsub.Subscription, time
 		b.Fatal("nMessages must be divisible by # of sending/receiving goroutines")
 	}
 	b.ResetTimer()
+	b.StopTimer()
 	for i := 0; i < b.N; i++ {
-		if !timeSend {
-			b.StopTimer()
-		}
-		if err := publishNConcurrently(topic, nMessages, concurrencySend, attrs, body); err != nil {
-			b.Fatalf("publishing: %v", err)
-		}
-		b.Logf("published %d messages", nMessages)
-		if timeSend {
-			b.StopTimer()
-		} else {
-			b.StartTimer()
-		}
-		if err := receiveNConcurrently(sub, nMessages, concurrencyReceive); err != nil {
-			b.Fatalf("receiving: %v", err)
+		var eg errgroup.Group
+		eg.Go(func() error {
+			if !timeSend {
+				b.StartTimer()
+				defer b.StopTimer()
+			}
+			b.Logf("publishing %d messages", nMessages)
+			if err := publishNConcurrently(topic, nMessages, concurrencySend, attrs, body); err != nil {
```

Comment:
Doesn't this still assume persistence, since this could happen before `Receive` is called below?

Reply:
Now that you mention it I have to admit this wasn't a test-driven change. I suspect the way Derek wrote the NATS driver it will work as long as the local subscription has been created, with messages queueing up locally, but need to check this...